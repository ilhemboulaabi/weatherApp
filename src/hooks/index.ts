import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux'
import type { RootState, AppDispatch } from '../store/store'

export const useHookDispatch: () => AppDispatch = useDispatch
export const useHookSelector: TypedUseSelectorHook<RootState> = useSelector