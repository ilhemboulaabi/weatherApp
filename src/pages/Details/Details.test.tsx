import { shallow, ShallowWrapper } from 'enzyme';
import DetailsPage from './Details';
import { useHookSelector } from '../../hooks';

jest.mock('react-redux', () => ({
  useSelector: jest.fn()
}));

jest.mock('react-router-dom', () => ({
  useNavigate: jest.fn(),
}));

describe('DetailsPage', () => {
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    (useHookSelector as jest.Mock).mockReturnValue({
      weather: {
        cityDetails: {
          latitude: 40.7128,
          longitude: -74.0060,
          hourly: [],
          hourly_units: {
            time: '',
            temperature_2m: ''
          }
        }
      }
    });

    wrapper = shallow(<DetailsPage />);
  });

  afterEach(() => {
    jest.resetAllMocks();
  });

  it('renders the component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
