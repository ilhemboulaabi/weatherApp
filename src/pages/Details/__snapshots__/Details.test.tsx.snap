// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsPage renders the component correctly 1`] = `ShallowWrapper {}`;
